### PR TITLE
Oh there's a bunch of stuff here. Man.

### PR DESCRIPTION
**Pull request by mariogiov**
The python pipeline is still WIP.

### CODE REVIEW

File: bin_types_by_size.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    # Get feature types from GFF file
+    if not feature_type:
+        gff = HTSeq.GFF_Reader(ann_file)
+        feature_types = sorted(collections.Counter((f.type for f in gff)))
+    else:
+        feature_types = [feature_type]
+    feature_counts = {}
+    # Annotate
+    for feature_type in feature_types:
+        # TODO here it probably makes more sense to pull the gff file into memory to keep from reloading it each time
```
Comment (guillermo-carrasco) [inline bin_types_by_size.py:35]: agree (if a gff file is not too big)

File: bin_types_by_size.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    gff = HTSeq.GFF_Reader(gff_file)
+    features_dict = HTSeq.GenomicArrayOfSets( "auto", stranded=False )
+    for feature in gff:
+        if feature.type == "feature_type":
+            features_dict[ feature.iv ] += feature.name
+    return features_dict
+
+def annotate_count_aln(aln_file, ann_file, output_dir, feature_type):
+    """
+    Annotated some alignments, and count that shit
```
Comment (guillermo-carrasco) [inline bin_types_by_size.py:60]: :no_mouth:

File: bin_types_by_size.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        with redirect_stream(sys.stdout, null):
+            try:
+                # Usage: count_reads_in_features( sam_filename, gff_filename, stranded, overlap_mode, feature_type, id_attribute, quiet, minaqual, samout )
+                return counts.count_reads_in_features(aln_file, ann_file, False, "union", feature_type, 'ID', False, 10, "")
+            except ValueError:
+            # Skip features that give errors due to e.g. incorrect formatting, missing attributes
+                return None
+
+
+class RedirectStream(object):
```
Comment (guillermo-carrasco) [inline bin_types_by_size.py:80]: Nice! :+1:

**Comment by guillermo-carrasco** [comment]
Awesome @mariogiov ! Just try to use logging instead of prints :-)

File: bin_types_by_size.py (hunk 2 above)
Comment (mariogiov) [inline bin_types_by_size.py:60]: [Explicit is better than implicit](http://www.python.org/dev/peps/pep-0020/)

**Pull request merged**